api/src/Entity/ScheduleEntry.php

```diff
@@ -185,12 +185,18 @@ public function getEmbeddedActivity(): ?Activity {
      */
     #[ApiProperty(writable: false, example: '/days/1a2b3c4d')]
     #[Groups(['read'])]
-    public function getDay(): Day {
+    public function getDay(): Day|null {
         $dayNumber = $this->getDayNumber();
 
-        return $this->period->days->filter(function (Day $day) use ($dayNumber) {
+        $filteredDays = $this->period->days->filter(function (Day $day) use ($dayNumber) {
             return $day->getDayNumber() === $dayNumber;
-        })->first();
+        });
+
+        if ($filteredDays->isEmpty()) {
+            return null;
+        }
```

Comment:
Ist das nicht in der alten Implementation bereits enthalten? Was passiert wenn man auf einer leeren Doctrine Collection `->first()` aufruft?

Reply:
`->first()` gibt bei einer leeren Colleciton `false` zurück, was wiederum ein 500er schmeisst, weil der Rückgabewert von Typ `Day` erwartet wird.

Sollte ja eigentlich nie auftauchen, weil das Day-Objekt ja da sein muss. War aber halt in den Testdaten nicht drin, weil die Seeds nicht komplett waren.

Von mir aus auch ok, wenn wir den Fehlerfall lassen (oder bewusst einen Error schmeissen) und dann via Sentry abfangen. Was meinst?

Reply:
Ja, bewusst einen Fehler werfen fände ich fast besser, weil wir dann schon näher an der Quelle des Problems (im Backend) sind mit der Sentry-Meldung. So wie dus jetzt hast kommt dann eine Sentry-Meldung vom Frontend das sich beklagt dass .day `null` ist, dann muss man erst mal nachsehen woher das denn kommt.

Reply:
Implemented with https://github.com/ecamp/ecamp3/pull/2334/commits/4a2d987e2472c1015649395451fa7352c690d332